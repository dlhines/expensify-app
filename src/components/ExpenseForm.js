import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';

// This little script will take any date and convert it to day/time format of the string
// const now = moment().format('MMM Do YYYY H:m:ss')
// const then = moment(1234567890);
// console.log(then._d)

export default class ExpenseForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            description: props.expense ? props.expense.description : '',
            note: props.expense ? props.expense.note : '',
            amount: props.expense ? (props.expense.amount / 100).toString() : '',
            createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
            calendarFocused: false,
            error: '',
        };        
    }
    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description }));
    };
    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState(() => ({ note }));
    }; 
    onAmountChange = (e) => {
        const amount = e.target.value;
        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(() => ({ amount }))
        } else {
            return this.state.amount;
        }
    };
    onDateChange = (createdAt) => {
        if (createdAt){
            this.setState(() => ({ createdAt }));
        }
    };
    onFocusChange = ({ focused }) => {
        this.setState(() => ({ calendarFocused: focused }));
    }
    onSubmit = (e) => {
        e.preventDefault();
        if (!this.state.description || !this.state.amount) {
            this.setState(() => ({error: "A description and amount are both required."}));
        } else {
            this.setState(() => ({error: ''}));
            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount, 10) * 100,
                createdAt: this.state.createdAt.valueOf(),
                note: this.state.note,
            });
        ;}
    };
    render() {
        return (
            <div> 
                <p>{this.state.error && this.state.error}</p>
                <form onSubmit={this.onSubmit}>
                    <input type="text" 
                        placeholder="Description" 
                        value={this.state.description} 
                        onChange={this.onDescriptionChange} 
                        autoFocus
                    />
                    <input 
                        type="text" 
                        placeholder="Amount" 
                        value={this.state.amount}
                        onChange={this.onAmountChange}
                    />
                    <SingleDatePicker
                        date={this.state.createdAt}
                        onDateChange={this.onDateChange}
                        focused={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        enableOutsideDays={true}
                        isOutsideRange={() => false}
                    />
                    <textarea 
                        placeholder="Add a note for your expense: (Optional)" 
                        onChange={this.onNoteChange}
                        value={this.state.note} 
                    >
                    </textarea>
                    <button>Add Expense</button>
                </form>
            </div>
        )
    }
}

