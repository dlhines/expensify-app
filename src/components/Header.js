import React from 'react';
import { NavLink, Link } from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <nav>
            <NavLink to="/" activeClassName="is-active" exact={true}>Expensify Dashboard</NavLink>        
            <NavLink to="/create" activeClassName="is-active">Add An Expense</NavLink>        
        </nav>    
    </header>
);

export default Header;