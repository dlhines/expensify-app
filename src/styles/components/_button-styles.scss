.box-layout__login_button {
    color: #EEE;
    background: lighten($buttonbg, 20%);
    border: none;
    font-size: $large;
    padding: $s-size;
}

.button {
    color: #EEE;
    display: inline-block;
    background: lighten($buttonbg, 20%);
    border: none;
    font-size: $large;
    padding: $s-size;
    text-decoration: none;
    line-height: 1;  
}

.button--link {
    background: none;
    color: #FFF;
}

.button--secondary {
    background: #888888;
    color: #000;
}