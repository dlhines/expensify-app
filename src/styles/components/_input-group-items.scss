.input-group {
    display: flex;
    flex-direction: column;
    justify-content: space-between;

    @media (min-width: $desktop-bp) {
        flex-direction: row;
        margin-bottom: $l-size;
    }
}

.input-group__item {
    margin-bottom: $s-size;
    @media (min-width: $desktop-bp) {
        flex-direction: row;
        margin-bottom: $m-size;
    }
}
