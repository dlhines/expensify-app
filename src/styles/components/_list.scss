.list-header{
    background: #f7f7f7;
    border: 1px solid grey;
    display: flex;
    justify-content: space-between;
    padding: $s-size $m-size;
}

.list-item {
    border: 1px solid #EEE;
    border-top: 0;
    color: #000;
    display: flex;
    flex-direction: column;
    padding: $s-size;
    text-decoration: none;
    transition: background .3s ease;
    &:hover {
        background: #FFF;
    } 
    @media (min-width: $desktop-bp) {
        align-items: center;
        flex-direction: row;
        justify-content: space-between;
        padding: $m-size;
    }    
}

.list-item--message {
    align-items: center;
    color: grey;
    justify-content: center;
    padding: $m-size;
}

.list-item__title {
    margin: 0;
    word-break: break-all;
}

.list-item__sub-title {
    color: grey;
    font-size: 1.2rem;
    margin-top: 1rem;
}

.list-item__data {
    margin: $s-size 0 0 0;
    @media (min-width: $desktop-bp) {
        margin: 0;
        padding-left: $s-size;
    }     
}