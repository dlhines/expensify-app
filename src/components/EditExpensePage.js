import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpenses, startRemoveExpense } from '../actions/expenses.js'

export class EditExpensePage extends React.Component {
    editExpense = (expense) => {
        this.props.startEditExpenses(this.props.expense.id, expense);
        this.props.history.push("/");
    };
    onRemove = (e) => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push("/");
    };
    render() {
        return (
        <div>
            <ExpenseForm expense={this.props.expense} onSubmit={this.editExpense} />
            <button onClick={this.onRemove}>Remove Item</button>              
        </div>
        );
    };    
}

const mapStateToProps = (state, props) => {
    return  {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id ),
    }
}

const mapDispatchToProps = (dispatch, props) => ({
    startEditExpenses: (id, expense)=> dispatch(startEditExpenses(id, expense)),
    startRemoveExpense: (id) => dispatch(startRemoveExpense(id)),
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);