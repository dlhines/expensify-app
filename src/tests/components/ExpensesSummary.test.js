import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={12} />);
    expect(wrapper).toMatchSnapshot();

});

test('should correctly render ExpensesSummary with multi expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={14} expensesTotal={12} />);
    expect(wrapper).toMatchSnapshot();
});