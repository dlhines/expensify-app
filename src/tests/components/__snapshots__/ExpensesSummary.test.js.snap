// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $0.12
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with multi expense 1`] = `
<div>
  <h1>
    Viewing 
    14
     
    expenses
     totalling 
    $0.12
  </h1>
</div>
`;
